lib/travis/build/script/dart.rb

```diff
@@ -87,62 +147,114 @@ def announce
         end
 
         def install
-          sh.if '-f pubspec.yaml' do
-            sh.cmd "pub get"
+          sh.fold 'pub_get' do
+            sh.echo 'pub get'
+
+            sh.if '-f pubspec.yaml' do
+              sh.cmd "pub get"
+            end
           end
         end
 
         def script
           # tests with test package
-          sh.if '[[ -d packages/test ]] || grep -q ^test: .packages 2> /dev/null', raw: true do
-            if with_content_shell
+          sh.if package_installed?('test'), raw: true do
+            if config[:with_content_shell]
               sh.export 'DISPLAY', ':99.0'
               sh.cmd 'sh -e /etc/init.d/xvfb start'
               # give xvfb some time to start
               sh.cmd 't=0; until (xdpyinfo -display :99 &> /dev/null || test $t -gt 10); do sleep 1; let t=$t+1; done'
               sh.cmd 'pub run test -p vm -p content-shell -p firefox'
             else
-              sh.cmd 'pub run test'
+              pub_run_test
             end
           end
           # tests with test_runner for old tests written with unittest package
-          sh.elif '[[ -d packages/unittest ]] || grep -q ^unittest: .packages 2> /dev/null', raw: true do
+          sh.elif package_installed?('unittest'), raw: true do
+            sh.fold "deprecated.unittest" do
+              sh.deprecate <<MESSAGE
+DEPRECATED: The unittest package is deprecated. Please upgrade to the test
+package. See https://github.com/dart-lang/test.
```

Comment:
I think the GitHub version is a lot more readable. I've updated the link to anchor to the README, though.